src/tools/nft/utils.ts

```diff
@@ -0,0 +1,208 @@
+      metadata: nft.metadata || {},
+    };
+  });
+}
+
+/**
+ * Detect if a contract is ERC721 or ERC1155 by checking if it supports the respective interface ID
+ * @param wallet Viem wallet client with public actions
+ * @param contractAddress The contract address to check
+ * @returns The detected NFT standard or "UNKNOWN"
+ */
+export async function detectNftStandard(
+  wallet: PublicActions,
+  contractAddress: string,
+): Promise<'ERC721' | 'ERC1155' | 'UNKNOWN'> {
+  try {
+    // ERC721 interface ID: 0x80ac58cd
+    const isErc721 = await wallet.readContract({
+      address: contractAddress as `0x${string}`,
+      abi: erc721Abi,
+      functionName: 'supportsInterface',
+      args: ['0x80ac58cd'],
+    });
+
+    if (isErc721) {
+      return 'ERC721';
+    }
+
+    // ERC1155 interface ID: 0xd9b67a26
+    const isErc1155 = await wallet.readContract({
+      address: contractAddress as `0x${string}`,
+      abi: erc1155Abi,
+      functionName: 'supportsInterface',
+      args: ['0xd9b67a26'],
+    });
+
+    if (isErc1155) {
+      return 'ERC1155';
+    }
+
+    return 'UNKNOWN';
+  } catch (error) {
+    console.error(
+      `Error detecting NFT standard for ${contractAddress}:`,
+      error,
+    );
+    return 'UNKNOWN';
+  }
+}
+
+/**
+ * Helper function to fetch NFTs from Alchemy API
+ * @param params Parameters for fetching NFTs
+ * @param params.ownerAddress The address to fetch NFTs for
+ * @param params.limit Maximum number of NFTs to fetch
+ * @returns The NFT data from Alchemy API
+ */
+export async function fetchNftsFromAlchemy(params: {
+  ownerAddress: string;
+  limit?: number;
+}): Promise<Record<string, unknown>> {
+  const { ownerAddress, limit = 50 } = params;
+  // Access environment variables safely
+  const apiKey =
+    typeof process !== 'undefined' ? process.env.ALCHEMY_API_KEY : undefined;
+
+  if (!apiKey) {
+    throw new Error('ALCHEMY_API_KEY is not set in environment variables');
+  }
+
+  try {
+    const baseUrl = 'https://base-mainnet.g.alchemy.com/nft/v3';
+    const url = `${baseUrl}/${apiKey}/getNFTsForOwner?owner=${ownerAddress}&withMetadata=true&pageSize=${limit}`;
+
+    const response = await fetch(url);
+
+    if (!response.ok) {
+      const errorText = await response.text();
+      console.error(`Alchemy API error (${response.status}): ${errorText}`);
+      throw new Error(
+        `Alchemy API error: ${response.status} ${response.statusText}`,
+      );
+    }
+
+    const data = await response.json();
+    return data;
+  } catch (error) {
+    console.error(`Error fetching NFTs from Alchemy:`, error);
+    throw error;
+  }
+}
+
+/**
+ * Transfer an NFT from one address to another
+ * @param params Parameters for the transfer
+ * @returns Transaction hash
+ */
+export async function transferNft(params: {
+  wallet: WalletClient & PublicActions;
+  contractAddress: string;
+  tokenId: string;
+  toAddress: string;
+  amount?: string;
+}): Promise<`0x${string}`> {
```

Comment:
We prefer params done like this:
```suggestion
type TransferNftParams = {
  wallet: WalletClient & PublicActions;
  contractAddress: string;
  tokenId: string;
  toAddress: string;
  amount?: string;
}

export async function transferNft({
  wallet,
  contractAddress,
  tokenId,
  toAddress,
  amount
}: TransferNftParams): Promise<`0x${string}`> {
```